resources/views/layouts/block.blade.php

```diff
@@ -1,8 +1,18 @@
 @php
 $print = app('printservice')->isPrintMode();
+$pClass = 'App\Http\Controllers\GenericPagesController';
+if (env('APP_ENV') != 'testing') {
+    $action = request()->route()->getAction();
+    if (is_array($action) && $action['controller']) {
+        $pClass = $action['controller'];
+    }
+}
+$pClass = preg_replace('/App\\\\Http\\\\Controllers\\\\/i','p-',$pClass);
+$pClass = preg_replace('/Controller/i','',$pClass);
+$pClass = strtolower(preg_replace('/@/i','-',$pClass));
```

Comment:
Instead of parsing this all out with regex, consider using the `getControllerClass()` and `getControllerMethod()` methods from the route:
```php
$route = request()->route();
$controller = $route->getControllerClass();
$method = $route->getControllerMethod();
$pClass = class_basename($controller)->before('Controller')->prepend('p-')->append('-')->append($method);
```

See https://laravel.com/api/10.x/Illuminate/Routing/Route.html#method_getControllerClass